backend/src/main/kotlin/blue/mild/covid/vaxx/auth/JwtService.kt

```diff
@@ -0,0 +1,87 @@
+package blue.mild.covid.vaxx.auth
+
+import blue.mild.covid.vaxx.dao.UserRole
+import blue.mild.covid.vaxx.dto.JwtConfigurationDto
+import blue.mild.covid.vaxx.dto.response.JwtResponseDtoOut
+import com.auth0.jwt.JWT
+import com.auth0.jwt.JWTCreator
+import com.auth0.jwt.JWTVerifier
+import com.auth0.jwt.algorithms.Algorithm
+import io.ktor.auth.jwt.JWTCredential
+import mu.KLogging
+import pw.forst.tools.katlib.toUuid
+import java.util.Date
+
+/**
+ * Service for issuing JWTs.
+ */
+class JwtService(
+    private val algorithm: Algorithm,
+    private val jwtConfiguration: JwtConfigurationDto
+) {
+
+    private companion object : KLogging() {
+        const val PUBLIC_SUBJECT = "public"
+        const val ROLE = "role"
+        const val TYPE = "type"
+    }
+
+    /**
+     * Build JWT verifier for JWTs issued by this service.
+     */
+    fun makeJwtVerifier(): JWTVerifier = JWT
+        .require(algorithm)
+        .withIssuer(jwtConfiguration.issuer)
+        .withAudience(jwtConfiguration.audience)
+        .build()
+
+    /**
+     * Generate and sign JWT for given principal.¬
```

Comment:
Co to je za znak na konci?

Reply:
asi preklep, macos klavesnice ma vselijake znaky na neobvyklych mistech